frontend/automation/cypress/support/commands.js

```diff
@@ -1,9 +1,87 @@
 require('cypress-downloadfile/lib/downloadFileCommand')
 
+const axios = require('axios')
+
 Cypress.Commands.add('getByTestId', (selector, ...args) => {
   return cy.get(`[data-testid=${selector}]`, ...args)
 })
 
 Cypress.Commands.add('clean', () => {
   return cy.exec('npm run clean')
 })
+
+Cypress.Commands.add('restoreWallet', () => {
+  const passphrase = '123'
+
+  // Visit a page so that the window object is bootstrapped with backend functions
+  cy.visit('#/wallet')
+
+  cy.window()
+    // Init wallet with local vega home
+    .then(win => {
+      const handler = win.go.backend.Handler
+
+      // First initialise app with local frontend directory
+      return (
+        handler
+          .InitialiseApp({
+            vegaHome: './frontend/automation/test-wallets'
+          })
+          // Import wallet using known recovery phrase setting
+          .then(() => {
+            return handler.ImportWallet({
+              wallet: 'test',
+              recoveryPhrase:
+                'behave unveil treat stone forward priority shoulder output woman dinner wide oval once fire move perfect together sail hero local try cinnamon clip hawk',
+              version: 2,
+              passphrase
+            })
+          })
+          // Store env vars and import a network
+          .then(res => {
+            // Store env vars for later use in tests and then import a network
+            Cypress.env('testWalletPassphrase', passphrase)
+            Cypress.env('testWalletName', res.wallet.name)
+            Cypress.env('testWalletPublicKey', res.key.publicKey)
+
+            return handler.ImportNetwork({
+              url: Cypress.env('testnetConfigUrl'),
+              name: 'fairground'
+            })
+          })
+      )
```

Comment:
chore (non-blocking0): can this be rewritten with async/await?

Reply:
the body of the function can be converted, but not the initial cy.window() call 👍